recipes/default.rb

```diff
@@ -15,3 +15,166 @@
+include_recipe 'perl'
+
+package %w(request-tracker mutt procmail)
+
+# Root Account
+template '/root/.rtrc' do
+  source 'rt/rtrc.erb'
+  cookbook 'osl-rt'
+  mode '0600'
+  sensitive true
+  variables(root_pass: node['osl-rt']['root-password'])
+end
+
+# Set up user for mail
+user node['osl-rt']['default'] do
+  manage_home true
+end
+
+# User defined Hostalias file in order to patch into the RT site with the RT CLI/procmail
+['root', "/home/#{node['osl-rt']['default']}"].each do |file_path|
+  file "/#{file_path}/.rthost" do
+    content <<~EOF
+      rtlocal localhost
+    EOF
+  end
+end
+
+# Add the RT command to the root user's PATH
+link '/usr/local/sbin/rt' do
+  to '/opt/rt/bin/rt'
+end
+
+# RT Initial Configuration.
+file '/opt/rt/etc/RT_SiteConfig.pm' do
+  # Use the init function in order to generate the perl config file
+  content osl_rt_init_config
+  group 'apache'
+  mode '0640'
+  sensitive true
+  notifies :reload, 'apache2_service[osuosl]'
+end
+
+# Initalize the DB
+execute 'init-db-rt' do
+  command <<-EOC
+    /opt/rt/sbin/rt-setup-database \
+      --action init \
+      --dba #{node['osl-rt']['db']['username']} \
+      --dba-password #{node['osl-rt']['db']['password']} \
+      --skip-create && \
+    touch /opt/rt/chef/init-db-rt
+  EOC
+  creates '/opt/rt/chef/init-db-rt'
+  sensitive true
+end
+
+# Set a new password for root
+execute 'Set root password' do
+  command <<-EOC
+    mysql -u #{node['osl-rt']['db']['username']} \
+      -p#{node['osl-rt']['db']['password']} \
+      -e 'UPDATE Users \
+        SET Password=md5(\"#{node['osl-rt']['root-password']}\") \
+        WHERE Name=\"root\";' \
+      #{node['osl-rt']['db']['name']} && \
+    touch /opt/rt/chef/init-root-passwd
+  EOC
+  creates '/opt/rt/chef/init-root-passwd'
+  sensitive true
+end
+
+# Set up web app
+apache_app node['osl-rt']['fqdn'] do
+  directory '/opt/rt/share/html'
+  include_config true
+  include_template true
+  include_directory 'rt'
+  include_name 'rt'
+  include_params('domain': node['osl-rt']['fqdn'])
+  server_aliases ['rtlocal']
+end
+
+# Forcefully reload Apache during the initial run, in order to allow for setting up the queues properly.
+# apache_app does not reload httpd after being ran, meaning the website is unavailable until after the converge has finished.
+service 'httpd' do
+  action :reload
+  not_if { ::File.exist?('/var/spool/mail/nobody') }
+end
+
+# Set up the queues in RT
+node['osl-rt']['queues'].each do |pt, email|
+  execute "Creating RT queue for #{pt}" do
+    command <<-EOC
+    HOSTALIASES=/root/.rthost \
+    /opt/rt/bin/rt create -t queue set \
+      name="#{pt}" correspondaddress="#{email}@#{node['osl-rt']['fqdn']}" \
+      commentaddress="#{email}-comment@#{node['osl-rt']['fqdn']}" \
+      && touch /tmp/#{email}done
+    EOC
+    creates "/tmp/#{email}done"
+  end
+end
+
+# Nobody mail directory
+file '/var/spool/mail/nobody' do
+  owner 'nobody'
+  group 'mail'
+  mode '0660'
+end
```

Comment:
Why does this need to be created?

Reply:
This was a leftover from the `osl-admin` transition. It's original goal was to be the end of the line for the RT mail that had no recipient. But that has since been changed to be the base user of the VM (`almalinux`, `centos`).

Reply:
What did we have it set in osl-admin?